src/compiler/parser.ts

```diff
@@ -1089,8 +1089,22 @@ namespace ts {
             return currentToken = scanner.reScanSlashToken();
         }
 
-        function reScanTemplateToken(): SyntaxKind {
-            return currentToken = scanner.reScanTemplateToken();
+        function reScanTemplateToken(isTaggedTemplate?: boolean): SyntaxKind {
+            if (isTaggedTemplate) {
+                let lastError: DiagnosticWithLocation | undefined;
+                while ((lastError = lastOrUndefined(parseDiagnostics)) && scanner.getTokenPos() < lastError.start) {
+                    parseDiagnostics.pop();
+                }
+            }
+            return currentToken = scanner.reScanTemplateToken(isTaggedTemplate);
+        }
+
+        function reScanTemplateHeadOrNoSubstitutionTemplate(): SyntaxKind {
+            let lastError: DiagnosticWithLocation | undefined;
+            while ((lastError = lastOrUndefined(parseDiagnostics)) && scanner.getTokenPos() < lastError.start) {

```

Comment:
Oof - is there really no way to avoid errors from the prior parse? This seems wrong, but is there any existing precedent for this?